### PR TITLE
Modification of Coverage Calculation

### PR DESCRIPTION
**Pull request by HenryDykhne**
# Modification of Coverage Calculation:
* Changed calculation of coverage score to be more accurate, taking into account coverage probabilities. Now, the total coverage of all constellations is no longer exactly the same.
* Added a more comprehensive printout of the solution found by the quantum computer, including coverage scores of each constellation, the total solution score, and a normalized solution score (1 being perfect. 0 being the same as not launching any satellites.).
* Updated the readme to remove mentions of average in coverage calculation.

# Files changed: 
* `README.md`
* `satellite.py`

### CODE REVIEW

File: satellite.py [hunk 1]
```diff
@@ -40,6 +40,14 @@
 import neal
 import sys
 from dwave.system import LeapHybridSampler
+# Pr(at least one event)=1−Pr(none of the events)
```
Comment (mcfarljm) [inline satellite.py]: For clarity, suggest noting that this applies to independent events:
```suggestion
# For independent events, Pr(at least one event)=1−Pr(none of the events)
```

Comment (mcfarljm) [inline satellite.py]: Minor style nitpick: please add two blank lines ahead of this function definition (https://www.python.org/dev/peps/pep-0008/#blank-lines)
```suggestion


# Pr(at least one event)=1−Pr(none of the events)
```

File: satellite.py [hunk 2]
```diff
@@ -40,6 +40,14 @@
...
+# https://math.stackexchange.com/questions/85849/calculating-the-probability-that-at-least-one-of-a-series-of-events-will-happen
+def calculateScore(constellation, data):
```
Comment (mcfarljm) [inline satellite.py]: Another style nitpick, but can you please rename the function to `calculate_score` to match our usual style?
https://www.python.org/dev/peps/pep-0008/#function-and-variable-names

**Review by mcfarljm — COMMENTED** [review]
Looks good, thanks for the contribution.  I have a few minor suggestions, mostly style related.  To make the edits, you can "accept" proposed changes in GitHub where applicable; otherwise, you can simply create and push new commits to the same branch, and GitHub will automatically add the new changes to this PR.

Also, due to the change in output format, this breaks our current unit tests.  My suggestion is to rework the unit tests as follows:
```diff
diff --git a/tests/test_satellite.py b/tests/test_satellite.py
index 5449e9e..115413d 100644
--- a/tests/test_satellite.py
+++ b/tests/test_satellite.py
@@ -21,17 +21,6 @@ example_dir = os.path.dirname(os.path.dirname(os.path.abspath(__file__)))
 
 
 class TestSmoke(unittest.TestCase):
-    # test that the example runs without failing
-    def test_smoke(self):
-        file_path = os.path.join(example_dir, 'satellite.py')
-        data_path = os.path.join(example_dir, 'small.json')
-        solver = 'neal'
-
-        value = subprocess.check_output([sys.executable, file_path, data_path, solver])
-
-        for constellation in eval(value):
-            self.assertIsInstance(constellation, frozenset)
-
     def test_satellite(self):
         """Verify contents of the output"""
 
@@ -40,14 +29,16 @@ class TestSmoke(unittest.TestCase):
         solver = 'neal'
 
         output = subprocess.check_output([sys.executable, file_path, data_path, solver])
-        output = str(output).upper()
+        output = output.decode('utf-8') # Bytes to str
+        output = output.lower()
 
         if os.getenv('DEBUG_OUTPUT'):
             print("Example output \n" + output)
 
-        with self.subTest(msg="Verify if output contains '[frozenset({' \n"):
-            self.assertIn("[frozenset({".upper(), output)
+        with self.subTest(msg="Verify output contains expected keywords \n"):
+            self.assertIn("constellation", output)
+            self.assertIn("score", output)
         with self.subTest(msg="Verify if error string contains in output \n"):
-            self.assertNotIn("ERROR", output)
+            self.assertNotIn("error", output)
         with self.subTest(msg="Verify if warning string contains in output \n"):
-            self.assertNotIn("WARNING", output)
+            self.assertNotIn("warning", output)
```

If you prefer, I can add this change via a separate PR, and then you can merge that or rebase your branch once it is ready.  Otherwise, you can add the above change directly as part of this PR.

File: satellite.py (hunk 1 above)
Comment (HenryDykhne) [reply satellite.py]: done

File: satellite.py (hunk 2 above)
Comment (HenryDykhne) [reply satellite.py]: done

**Comment by HenryDykhne** [comment]
changes made as requested @mcfarljm 
Please confirm at your leasure.

**Review by mcfarljm — APPROVED** [review]
Looks good to me, thanks for the improvement!

**Comment by HenryDykhne** [comment]
no problem. can you merge it? I don't have permission to do so. @mcfarljm

**Pull request merged**